lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -815,7 +818,17 @@ def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
         if self.opensearch.is_started():
             try:
                 self._post_start_init(event)
-            except (OpenSearchHttpError, OpenSearchNotFullyReadyError):
+            except (
+                OpenSearchHttpError,
+                OpenSearchStartTimeoutError,
+                OpenSearchNotFullyReadyError,
+            ):
+                event.defer()
+            except (OpenSearchStartError, OpenSearchUserMgmtError) as e:
+                # Either generic start failure or cluster is not read to create the internal users
+                logger.exception(e)
+                self.node_lock.release()
+                self.status.set(BlockedStatus(ServiceStartError))
```

Comment:
i'm not sure about this one - the service effectively started, the user management failure is unrelated to the start

Reply:
Hum, makes sense. I also want to remove that `logger.exception()` from the user error path.